packages/data-context/src/actions/MigrationActions.ts

```diff
@@ -189,8 +189,17 @@ export class MigrationActions {
         return this.ctx.onError(getError('MIGRATION_CYPRESS_NOT_FOUND'))
       }
 
-      if (!semver.satisfies(version, '^10.0.0')) {
-        return this.ctx.onError(getError('MIGRATION_MISMATCHED_CYPRESS_VERSIONS', version))
+      const currentVersion = (await this.ctx.versions.versionData()).current.version
+
+      // Validate that the project being migrated has a version of Cypress compatible with the version being executed.
+      // This handles situations where Cypress is launched in global mode to migrate a project with an older version of
+      // Cypress as a dependency which could break the project when launched directly.
+      // For example:
+      //    Local: 9.6.0     Global: 10.0.0     FAIL
+      //    Local: 10.0.1    Global: 10.0.0     PASS
+      //    Local: 12.0.0    Global: 12.0.1     FAIL
```

Comment:
It should be fine to have a global version greater than local as long as they are the same major version right?

Reply:
My thought was that a point release may add new config values that an older point release would fail to parse. For example, the Svelte support added in 10.7.0 introduced a `framework` value that a 10.6.0 version of Cypress wouldn't know how to handle. Theoretically a newer local version should handle of a config built by an older global version so long as they're the same major version

Reply:
More strict = more safety, I'm cool with that. Generally, if you're trying to migrate a project it should have the most up to date version of cypress available.